scroll_menu.py

```diff
@@ -7,45 +11,88 @@
     CompletionsMenu,
     Float,
     FloatContainer,
+    FormattedTextControl,
     HSplit,
     Layout,
     ScrollablePane,
     VSplit,
+    Window,
 )
-from prompt_toolkit.widgets import Frame, Label, TextArea
+from prompt_toolkit.widgets import Button, Frame, Label
+
+
+class ScrollableMenu:
+    """Main class for scrollable menu"""
+
+    def __init__(self, dir: str):
+        # Display contents in the current working directory on the menu
+        cwd_content = os.listdir(dir)
+        frames = [
+            Frame(
+                Button(
+                    text=item,
+                    handler=functools.partial(self.display_content, item, "."),
+                )
+            )
+            for item in cwd_content
+        ]
+
+        self.body = VSplit(
+            children=[
+                Label(text="File's content here"),
+                Frame(ScrollablePane(HSplit(children=frames))),
+            ],
+            padding_char="|",
+            padding=1,
+        )
+
+    def display_content(self, target_content: str, target_dir: str) -> None:
+        """Display content.
+
+        If target's content is a file, display it in the left column.
+        If the target_content is a directory, replace the right column with its content.
+
+        Args:
+            target_content (str): Target's content
+            target_dir (str): target's directory
+        """
+        if isfile(join(target_dir, target_content)):
+            # open file's content
+            with open(join(target_dir, target_content), "r") as f:
+                file_content = f.read()
+            # Remove any object that isn't HSplit
+            self.body.children = list(
+                filter(lambda x: type(x) == HSplit, self.body.children)
+            )
+            # Prepend the file_content to the body
+            self.body.children.insert(
+                0, Window(content=FormattedTextControl(file_content))
+            )
+        elif isdir(join(target_dir, target_content)):
+            pass
```

Comment:
I'm running this locally and clicking on a directory doesn't do anything.

It seems like that is intentional, should we add some placeholder functionality and add a Issue to implement that?

Reply:
From @CupOfGeo: We also need an ⬆️ move up on directory button for this use case.